src/aker_metrics.c

```diff
@@ -0,0 +1,241 @@
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_inc_window_trans_count()
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->window_trans_count += 1;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_inc_schedule_set_count()
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->schedule_set_count += 1;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_inc_md5_err_count()
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->md5_err_count += 1;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_set_process_start_time( time_t val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->process_start_time = val;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_set_schedule_enabled( int val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->schedule_enabled = val;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_set_tz( const char *val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	if( g_metrics->timezone )
+	{
+		free(g_metrics->timezone);
+	}
+
+	g_metrics->timezone = strdup(val);
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void aker_metric_set_tz_offset( long int val )
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	g_metrics->timezone_offset = val;
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+void stringify_metrics(int flag)
+{
+	pthread_mutex_lock(&aker_metrics_mut);
+
+	char str[512];
+	aker_metrics_t * tmp = NULL;
+	tmp = g_metrics;
+
+	if(g_metrics != NULL)
+	{
+		snprintf(str, 512, "DeviceBlockCount,%d,"
+                   "WindowTransistionCount,%d,"
+                   "ScheduleSetCount,%d,"
+                   "MD5ErrorCount,%d,"
+                   "ProcessStartTime,%ld,"
+                   "ScheduleEnabled,%d,"
+                   "TimeZone,%s,"
+                   "TimeZoneOffset,%+ld",
+
+                   tmp->device_block_count,
+                   tmp->window_trans_count,
+                   tmp->schedule_set_count,
+                   tmp->md5_err_count,
+                   tmp->process_start_time,
+                   tmp->schedule_enabled,
+                   tmp->timezone,
+                   tmp->timezone_offset);
+
+
+		debug_info("The stringified valued is (%s)\n", str);
+
+		if(flag)
+		{
+
+	#if defined(ENABLE_FEATURE_TELEMETRY2_0)
+			t2_event_s("akermetrics", str);
+			debug_info("akermetrics t2 event triggered\n");
+	#endif
+		}
+
+	}
+	else
+	{
+		debug_error("The g_metrics is NULL\n");
+	}
+
+	pthread_mutex_unlock(&aker_metrics_mut);
+}
+
+/* See aker_metrics.h for details. */
+int get_blocked_mac_count(const char* blocked)
```

Comment:
This isn't quite right since if the blocked string is NULL, the value of 1 is counted.  Doing memcmp() for 1 character is not ideal.  Finally, by casting `blocked` to `char*` removes the protection we added to the function call.  Don't do this.

Simpler would be:
```c
/* See aker_metrics.h for details. */
int get_blocked_mac_count(const char* blocked)
{
    int count = 0;
    const char *p = blocked;

    if(p) {
        /* If there are characters, then there there is 1+the number of spaces,
         * otherwise there are 0 blocked devices. */
        count = 1;

        while(*p) {
            if(' ' == *p) {
                count++;
            }
            p++;
        }
    }

	debug_info("the count is %d\n", count);
	debug_info("The mac after process were %s\n", blocked);
	return count;
}
```